hpke/kemBase.go

```diff
@@ -0,0 +1,241 @@
+package hpke
+
+import (
+	"crypto"
+	_ "crypto/sha256" // linking sha256 packages.
+	_ "crypto/sha512" // linking sha512 packages.
+	"encoding"
+	"encoding/binary"
+	"io"
+
+	"github.com/cloudflare/circl/kem"
+	"golang.org/x/crypto/hkdf"
+)
+
+type dhKEM interface {
+	sizeDH() int
+	calcDH(dh []byte, sk kem.PrivateKey, pk kem.PublicKey) error
+	GenerateKey() (kem.PublicKey, kem.PrivateKey, error)
+	DeriveKey(seed []byte) (kem.PublicKey, kem.PrivateKey)
+	UnmarshalBinaryPrivateKey(data []byte) (kem.PrivateKey, error)
+	UnmarshalBinaryPublicKey(data []byte) (kem.PublicKey, error)
```

Comment:
It seems like what we want is for objects of type `dhKEM` is to implement the `kem.Scheme` interface, plus some other DH-specific stuff. Why not replace this with `kem.Scheme`?

Reply:
there are two clases of kem in this package, both must implement `kem.Scheme` .
the dhKEM interface summarizes what is not common from these two implementations.
`  kem.Scheme = kemBase + dhKem `

Reply:
But unless you explicitly declare it, and object that implements `dhKEM` doesn't actually implement `kem.Scheme`. Put another way, I believe you can't coerce a `dhKEM` implementor into a `kem.Scheme` implementor.